distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso

```diff
@@ -442,6 +443,53 @@ type Number
      floating point numbers.
 @Builtin_Type
 type Float
+    ## GROUP Conversions
+       ICON convert
+       Generic conversion of an arbitrary Enso value to requested type.
+       Delegates to appropriate `.from` conversion method, if it exists.
+       If such method doesn't exist, `No_Such_Conversion` panic is raised.
+
+       Arguments:
+       - target_type: the requested type.
+
+       > Example
+         Following code defines conversion of a `Complex` type to a `Number`
+         by computing absolute distance from `0`. The code yields `5.0`:
+
+             type Complex
+                 Value re:Number im:Number
+
+             Number.from (that:Complex) = that.re*that.re+that.im*that.im . sqrt
+
+             Complex.Value 3 4 . to Number
+
+       > Example
+         `.from` conversion methods may have additional arguments
+         with default values. Thus the conversion from `Complex` to
+         `Number` may take additional argument:
+
+             type Complex
+                 Value re:Number im:Number
+
+             Number.from (that:Complex) = that.re*that.re+that.im*that.im . sqrt
+
+             Complex.Value 3 4 . to Number
+
+             type Complex
+                 Value re:Number im:Number
+
+             Number.from (that:Complex) (ignore_im:Boolean=False) = case ignore_im of
+                 False -> that.re*that.re+that.im*that.im . sqrt
+                 True -> that.re
+
+             yields_3 = Complex.Value 3 4 . to Number ignore_im=True
+             yields_5 = Complex.Value 3 4 . to Number ignore_im=False
+             default5 = Complex.Value 3 4 . to Number
+    to : Any -> Any ! No_Such_Conversion
+    to self target_type = case target_type of
+        Decimal -> Decimal.from self explicit=True
+        _ -> Any.to self target_type ...
```

Comment:
`Decimal.from` is used implicitly when we have typed arguments.
Following a chat with @GregoryTravis, we should remove the `explicit` parameter from the `Decimal.from` but this method should use `Decimal.from_float` directly.

This special case of `to` and `from` being asymmetric is needed for handling warnings / errors appropriately,

Reply:
In order to get the warnings and errors we want for various cases, we're violating the `to`/`from` symmetry slightly in this case. We are treating `from` as an implicit conversion, and `to` as an explicit conversion -- even though `from` can be used explicitly.

Implicit conversions result in warnings (e.g. for comparisons) or errors (for arithmetic), while explicit conversions do not.

I've added more documentation to make this clear. I've added the same documentation to each construction and conversion method, because it's important for users to know what their options are and what the consequences are. I've also fixed some inconsistencies in older methods and added tests.

The policy we have right now about when to give warnings and errors isn't set in stone; we'll have to use it and see how it works, and probably tweak it.